clang/lib/Gold/GoldParser.cpp

```diff
@@ -57,10 +65,13 @@ TokenKind CloseTokens[]
   tok::RightBracket,
   tok::Dedent,
   tok::Greater,
+  tok::HashGreater,
+  tok::Greater,
+  tok::Greater,
 };
 
 /// A class to help match enclosing tokens.
-template<EnclosureKind K>
+template<EnclosureKind K, bool NoFetchExpect = false>
```

Comment:
add to the comment what NoFetchExpect does. assuming it stops from fetching a token? in that case, ExpectNoFetch is a much clearer name